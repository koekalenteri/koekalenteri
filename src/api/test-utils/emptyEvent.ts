import type { ConfirmedEvent } from '../../types'

export const emptyEvent: ConfirmedEvent = {
  classes: [{ class: 'ALO' }],
  contactInfo: {
    official: {
      name: 'Teemu Toimitsija',
      email: 'official@example.com',
      phone: '040-official',
    },
    secretary: {
      name: 'Siiri Sihteeri',
      email: 'secretary@example.com',
      phone: '040-secretary',
    },
  },
  cost: 123,
  costMember: 123,
  createdAt: new Date('1989-01-01'),
  createdBy: 'test',
  description: 'test',
  endDate: new Date('1990-01-01'),
  entries: 0,
  entryEndDate: new Date('1990-01-01'),
  entryStartDate: new Date('1990-01-01'),
  eventType: 'test',
  id: 'test',
  judges: [123],
  kcId: 123456,
  location: 'test',
  modifiedAt: new Date('1989-01-02'),
  modifiedBy: 'test',
  name: 'test',
  official: {
    id: '1',
    name: 'Teemu Toimitsija',
  },
  priority: ['member'],
  secretary: {
    id: '2',
    name: 'Siiri Sihteeri',
  },
  organizer: {
    id: '1',
    name: 'Suomen Noutajakoirajärjestö ry',
  },
  places: 10,
  startDate: new Date('1990-01-01'),
  state: 'confirmed' as const,
}
