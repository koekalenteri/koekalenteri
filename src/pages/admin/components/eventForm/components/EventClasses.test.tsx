import { render, screen } from '@testing-library/react'
import { parseISO } from 'date-fns'

import { flushPromises, renderWithUserEvents } from '../../../../../test-utils/utils'

import EventClasses from './EventClasses'

const date = parseISO('2023-01-17')
const date2 = parseISO('2023-01-18')

describe('EventClasses', () => {
  beforeAll(() => jest.useFakeTimers())
  afterEach(() => jest.clearAllTimers())
  afterAll(() => jest.useRealTimers())

  it('should render with minimal properties', () => {
    const { container } = render(
      <EventClasses id={''} eventStartDate={date} eventEndDate={date} value={undefined} classes={[]} label={''} />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render with classes', () => {
    const { container } = render(
      <EventClasses
        id={''}
        eventStartDate={date}
        eventEndDate={date}
        value={undefined}
        classes={[{ class: 'ALO' }, { class: 'AVO' }]}
        label={''}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render with classes and value', () => {
    const { container } = render(
      <EventClasses
        id={''}
        eventStartDate={date}
        eventEndDate={date}
        value={[{ class: 'ALO', date, judge: { id: 1, name: 'Test Judge' } }]}
        classes={[
          { class: 'ALO', date },
          { class: 'AVO', date },
        ]}
        label={''}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render with classes and values', () => {
    const { container } = render(
      <EventClasses
        id={''}
        eventStartDate={date}
        eventEndDate={date}
        value={[
          { class: 'ALO', date, judge: { id: 1, name: 'Test Judge' } },
          {
            class: 'AVO',
            date,
            judge: [
              { id: 1, name: 'Test Judge' },
              { id: 2, name: 'Test Judge2' },
            ],
          },
        ]}
        classes={[
          { class: 'ALO', date },
          { class: 'AVO', date },
        ]}
        label={''}
        showCount
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render with classes and values, open', async () => {
    const { container, user } = renderWithUserEvents(
      <EventClasses
        id={''}
        eventStartDate={date}
        eventEndDate={date}
        value={[
          { class: 'ALO', date, judge: { id: 1, name: 'Test Judge' } },
          {
            class: 'AVO',
            date,
            judge: [
              { id: 1, name: 'Test Judge' },
              { id: 2, name: 'Test Judge2' },
            ],
          },
        ]}
        classes={[
          { class: 'ALO', date },
          { class: 'AVO', date },
        ]}
        label={''}
        showCount
      />,
      undefined,
      { advanceTimers: jest.advanceTimersByTime }
    )

    const input = screen.getByRole('combobox')
    await user.type(input, '{ArrowDown}')
    await flushPromises()

    expect(container).toMatchSnapshot()
  })

  it('should render with classes and values for 2 day event, open', async () => {
    const { container, user } = renderWithUserEvents(
      <EventClasses
        id={''}
        eventStartDate={date}
        eventEndDate={date2}
        value={[
          { class: 'ALO', date, judge: { id: 1, name: 'Test Judge' } },
          {
            class: 'AVO',
            date,
            judge: [
              { id: 1, name: 'Test Judge' },
              { id: 2, name: 'Test Judge2' },
            ],
          },
        ]}
        classes={[
          { class: 'ALO', date },
          { class: 'AVO', date },
          { class: 'AVO', date: date2 },
        ]}
        label={''}
        showCount
      />,
      undefined,
      { advanceTimers: jest.advanceTimersByTime }
    )

    const input = screen.getByRole('combobox')
    await user.type(input, '{ArrowDown}')
    await flushPromises()

    expect(container).toMatchSnapshot()
  })
})
