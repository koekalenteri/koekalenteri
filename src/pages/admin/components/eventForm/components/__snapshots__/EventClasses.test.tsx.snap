// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventClasses should render with classes 1`] = `
<div>
  <div
    class="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasPopupIcon css-1etx6i0-MuiAutocomplete-root"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-autocomplete="list"
          aria-expanded="false"
          aria-invalid="false"
          autocapitalize="none"
          autocomplete="off"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
          id=""
          role="combobox"
          spellcheck="false"
          type="text"
          value=""
        />
        <div
          class="MuiAutocomplete-endAdornment css-p1olib-MuiAutocomplete-endAdornment"
        >
          <button
            aria-label="Open"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
            tabindex="-1"
            title="Open"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;

exports[`EventClasses should render with classes and value 1`] = `
<div>
  <div
    class="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasPopupIcon css-1etx6i0-MuiAutocomplete-root"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-1gywuxd-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault MuiAutocomplete-tag MuiAutocomplete-tagSizeMedium css-n4kcu0-MuiChip-root"
          data-tag-index="0"
          tabindex="-1"
        >
          <span
            class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
          >
            ALO
          </span>
        </div>
        <input
          aria-autocomplete="list"
          aria-expanded="false"
          aria-invalid="false"
          autocapitalize="none"
          autocomplete="off"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-152mnda-MuiInputBase-input-MuiOutlinedInput-input"
          id=""
          role="combobox"
          spellcheck="false"
          type="text"
          value=""
        />
        <div
          class="MuiAutocomplete-endAdornment css-p1olib-MuiAutocomplete-endAdornment"
        >
          <button
            aria-label="Open"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
            tabindex="-1"
            title="Open"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;

exports[`EventClasses should render with classes and values 1`] = `
<div>
  <div
    class="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasPopupIcon css-1etx6i0-MuiAutocomplete-root"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-1gywuxd-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault MuiAutocomplete-tag MuiAutocomplete-tagSizeMedium css-n4kcu0-MuiChip-root"
          data-tag-index="0"
          tabindex="-1"
        >
          <span
            class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
          >
            ALO
          </span>
        </div>
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault MuiAutocomplete-tag MuiAutocomplete-tagSizeMedium css-n4kcu0-MuiChip-root"
          data-tag-index="1"
          tabindex="-1"
        >
          <span
            class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
          >
            AVO x2
          </span>
        </div>
        <input
          aria-autocomplete="list"
          aria-expanded="false"
          aria-invalid="false"
          autocapitalize="none"
          autocomplete="off"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-152mnda-MuiInputBase-input-MuiOutlinedInput-input"
          id=""
          role="combobox"
          spellcheck="false"
          type="text"
          value=""
        />
        <div
          class="MuiAutocomplete-endAdornment css-p1olib-MuiAutocomplete-endAdornment"
        >
          <button
            aria-label="Open"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
            tabindex="-1"
            title="Open"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;

exports[`EventClasses should render with classes and values for 2 day event, open 1`] = `
<div>
  <div
    aria-owns="-listbox"
    class="MuiAutocomplete-root Mui-expanded Mui-focused MuiAutocomplete-fullWidth MuiAutocomplete-hasPopupIcon css-1etx6i0-MuiAutocomplete-root"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-1gywuxd-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault MuiAutocomplete-tag MuiAutocomplete-tagSizeMedium css-n4kcu0-MuiChip-root"
          data-tag-index="0"
          tabindex="-1"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault MuiChip-avatar MuiChip-avatarSmall MuiChip-avatarColorDefault css-19awx12-MuiAvatar-root"
          >
            eeeeee
          </div>
          <span
            class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
          >
            ALO
          </span>
        </div>
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault MuiAutocomplete-tag MuiAutocomplete-tagSizeMedium css-n4kcu0-MuiChip-root"
          data-tag-index="1"
          tabindex="-1"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault MuiChip-avatar MuiChip-avatarSmall MuiChip-avatarColorDefault css-19awx12-MuiAvatar-root"
          >
            eeeeee
          </div>
          <span
            class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
          >
            AVO x2
          </span>
        </div>
        <input
          aria-activedescendant="-option-1"
          aria-autocomplete="list"
          aria-controls="-listbox"
          aria-expanded="true"
          aria-invalid="false"
          autocapitalize="none"
          autocomplete="off"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-152mnda-MuiInputBase-input-MuiOutlinedInput-input"
          id=""
          role="combobox"
          spellcheck="false"
          type="text"
          value=""
        />
        <div
          class="MuiAutocomplete-endAdornment css-p1olib-MuiAutocomplete-endAdornment"
        >
          <button
            aria-label="Close"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator MuiAutocomplete-popupIndicatorOpen css-z0g2be-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
            tabindex="-1"
            title="Close"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    class="base-Popper-root MuiAutocomplete-popper MuiAutocomplete-popperDisablePortal css-1gsu57t-MuiPopper-root-MuiAutocomplete-popper"
    data-popper-escaped=""
    data-popper-placement="bottom"
    data-popper-reference-hidden=""
    role="presentation"
    style="position: absolute; top: 0px; left: 0px; width: 0px; margin: 0px; transform: translate(0px, 0px);"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAutocomplete-paper css-9e5uuu-MuiPaper-root-MuiAutocomplete-paper"
    >
      <ul
        aria-labelledby="-label"
        class="MuiAutocomplete-listbox css-gdh49b-MuiAutocomplete-listbox"
        id="-listbox"
        role="listbox"
      >
        <li>
          <div
            class="MuiListSubheader-root MuiListSubheader-gutters MuiListSubheader-sticky MuiAutocomplete-groupLabel css-esv6et-MuiListSubheader-root-MuiAutocomplete-groupLabel"
          >
            eeeeee d.M.
          </div>
          <ul
            class="MuiAutocomplete-groupUl css-91wxkq-MuiAutocomplete-groupUl"
          >
            <li
              aria-disabled="false"
              aria-selected="true"
              class="MuiAutocomplete-option"
              data-option-index="0"
              id="-option-0"
              role="option"
              tabindex="-1"
            >
              <span
                class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                style="margin-right: 8px;"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate="false"
                  type="checkbox"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="CheckBoxIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2m-9 14-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              ALO
            </li>
            <li
              aria-disabled="false"
              aria-selected="true"
              class="MuiAutocomplete-option Mui-focused Mui-focusVisible"
              data-option-index="1"
              id="-option-1"
              role="option"
              tabindex="-1"
            >
              <span
                class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                style="margin-right: 8px;"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate="false"
                  type="checkbox"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="CheckBoxIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2m-9 14-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              AVO
            </li>
            <li
              aria-disabled="false"
              aria-selected="false"
              class="MuiAutocomplete-option"
              data-option-index="2"
              id="-option-2"
              role="option"
              tabindex="-1"
            >
              <span
                class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                style="margin-right: 8px;"
              >
                <input
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate="false"
                  type="checkbox"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="CheckBoxOutlineBlankIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5zm0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              AVO
            </li>
          </ul>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`EventClasses should render with classes and values, open 1`] = `
<div>
  <div
    aria-owns="-listbox"
    class="MuiAutocomplete-root Mui-expanded Mui-focused MuiAutocomplete-fullWidth MuiAutocomplete-hasPopupIcon css-1etx6i0-MuiAutocomplete-root"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-1gywuxd-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault MuiAutocomplete-tag MuiAutocomplete-tagSizeMedium css-n4kcu0-MuiChip-root"
          data-tag-index="0"
          tabindex="-1"
        >
          <span
            class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
          >
            ALO
          </span>
        </div>
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault MuiAutocomplete-tag MuiAutocomplete-tagSizeMedium css-n4kcu0-MuiChip-root"
          data-tag-index="1"
          tabindex="-1"
        >
          <span
            class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
          >
            AVO x2
          </span>
        </div>
        <input
          aria-activedescendant="-option-1"
          aria-autocomplete="list"
          aria-controls="-listbox"
          aria-expanded="true"
          aria-invalid="false"
          autocapitalize="none"
          autocomplete="off"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-152mnda-MuiInputBase-input-MuiOutlinedInput-input"
          id=""
          role="combobox"
          spellcheck="false"
          type="text"
          value=""
        />
        <div
          class="MuiAutocomplete-endAdornment css-p1olib-MuiAutocomplete-endAdornment"
        >
          <button
            aria-label="Close"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator MuiAutocomplete-popupIndicatorOpen css-z0g2be-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
            tabindex="-1"
            title="Close"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    class="base-Popper-root MuiAutocomplete-popper MuiAutocomplete-popperDisablePortal css-1gsu57t-MuiPopper-root-MuiAutocomplete-popper"
    data-popper-escaped=""
    data-popper-placement="bottom"
    data-popper-reference-hidden=""
    role="presentation"
    style="position: absolute; top: 0px; left: 0px; width: 0px; margin: 0px; transform: translate(0px, 0px);"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAutocomplete-paper css-9e5uuu-MuiPaper-root-MuiAutocomplete-paper"
    >
      <ul
        aria-labelledby="-label"
        class="MuiAutocomplete-listbox css-gdh49b-MuiAutocomplete-listbox"
        id="-listbox"
        role="listbox"
      >
        <li
          aria-disabled="false"
          aria-selected="true"
          class="MuiAutocomplete-option"
          data-option-index="0"
          id="-option-0"
          role="option"
          tabindex="-1"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            style="margin-right: 8px;"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
              data-testid="CheckBoxIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2m-9 14-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          ALO
        </li>
        <li
          aria-disabled="false"
          aria-selected="true"
          class="MuiAutocomplete-option Mui-focused Mui-focusVisible"
          data-option-index="1"
          id="-option-1"
          role="option"
          tabindex="-1"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            style="margin-right: 8px;"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
              data-testid="CheckBoxIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2m-9 14-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          AVO
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`EventClasses should render with minimal properties 1`] = `
<div>
  <div
    class="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasPopupIcon css-1etx6i0-MuiAutocomplete-root"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-autocomplete="list"
          aria-expanded="false"
          aria-invalid="false"
          autocapitalize="none"
          autocomplete="off"
          class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
          disabled=""
          id=""
          role="combobox"
          spellcheck="false"
          type="text"
          value=""
        />
        <div
          class="MuiAutocomplete-endAdornment css-p1olib-MuiAutocomplete-endAdornment"
        >
          <button
            aria-label="Open"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
            disabled=""
            tabindex="-1"
            title="Open"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;
